### DISCUSSION TITLE
Enter refreshes app with query params in url?

### DISCUSSION THREAD

**Discussion by ghost** [react-hook-form/react-hook-form#6738] [General] [answered]
Only one form in my app behaves this way and im not sure why? If I go inside an input and hit enter key the whole app refreshes and adds a query param to my url. None of my other forms do this and I am not sure why? 

Edit: Found a pattern. This only applies if the form only has one input. If it has multiple this issue does not occur.

**Comment by bluebill1049** [comment]
sounds like `e.preventDefault` issue

**Comment by ghost** [comment, marked as answer]
https://github.com/react-hook-form/react-hook-form/issues/6739#issuecomment-938190110

**Comment by mraj12** [comment]
same error the entire page refreshes but it behaves in Datepicker when i hit to Onchange it changes the query Params  and page refresh
note : useWatch to select range
